File: Tutorials/WPF/Classic/VB/DataAccess/BaseObject.vb
Imports DevExpress.Xpo

Namespace XpoTutorial

	<NonPersistent> _
	<DeferredDeletion, OptimisticLocking>
	Public Class BaseObject
		Inherits PersistentBase

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Key(True)> _
		<Persistent("OID")> _
		Private fOid As Integer
		<PersistentAlias("fOid")>
		Public Property Oid() As Integer
			Get
				Return fOid
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue(NameOf(Oid), fOid, value)
			End Set
		End Property
	End Class
End Namespace

File: Tutorials/WPF/Classic/CS/DataAccess/BaseObject.cs
using DevExpress.Xpo;

namespace XpoTutorial {

    [NonPersistent]
    [DeferredDeletion]
    [OptimisticLocking]
    public class BaseObject : PersistentBase {
        public BaseObject(Session session) : base(session) { }

        [Key(true)]
        [Persistent("OID")]
        int fOid;
        [PersistentAlias("fOid")]
        public int Oid {
            get { fOid; }
        }
    }
}
